oauth2client/dictionary_storage.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2014 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Dictionary storage for OAuth2 Credentials."""
+
+from oauth2client.client import OAuth2Credentials
+from oauth2client.locked_storage import LockedStorage
+
+
+__author__ = 'jonwayne@google.com (Jon Wayne Parrott)'
+
+
+class DictionaryStorage(LockedStorage):
```

Comment:
Hmm. I understand, but I'm unsure how to better solve this.

Basically the current case is that `oauth2client.client.Storage` defines `locked_{get,put,delete}` as abstract, and provides `{get,put,delete}` as public, which call `{acquire,release}_lock` as needed, but defines those methods as empty. That is to say that `Storage` seems undecided on whether it wants to provide locking. Ideally, I would just roll `LockedStorage` into `Storage`. This may be possible (I haven't tried), but could lead to weirdness with the already complicated `multistore_file`.

So I guess my question is which would you prefer:

* Move the few lines of code in `LockedStorage` into `client.Storage`, where honestly it kind of belongs.
* Duplicate locking logic in all of the sub-classes of `Storage`, which are planned to be numerous.
* Simplify `Storage` to abstract `{get,put,delete}`, move all locking stuff into `LockedStorage`.